internal/server/get_analysisrun_logs_v1alpha1.go

```diff
@@ -345,15 +345,22 @@ func (s *server) getStageFromAnalysisRun(
 	ctx context.Context,
 	run *rolloutsapi.AnalysisRun,
 ) (*kargoapi.Stage, error) {
-	stageName, ok := run.Labels[kargoapi.LabelKeyStage]
+	// The (full) name of the Stage is stored in the AnalysisRun's annotations
+	// when the label value contains a (hash-)shortened version of the Stage
+	// name. Check for this value first, and if it is not found, check the
+	// labels. If neither is found, return an error.
+	stageName, ok := run.Annotations[kargoapi.AnnotationKeyStage]
```

Comment:
As before this change, Stages with a name >63 characters were not possible at all. This change is fully backwards compatible.